src/WormholeRouter.sol

```diff
@@ -18,33 +18,26 @@ pragma solidity 0.8.9;
 
 import "./WormholeGUID.sol";
 
-interface WormholdJoinLike {
-    function registerWormholeAndWithdraw(WormholeGUID calldata wormholeGUID, uint256 maxFee) external;
-    function settle(bytes32 sourceDomain, uint256 batchedDaiToFlush) external;
-}
-
 interface TokenLike {
   function transferFrom(address _from, address _to, uint256 _value) external returns (bool success);
 }
 
-interface L1BridgeLike {
-    function initiateRequestMint(WormholeGUID calldata wormholeGUID, uint256 maxFee) external;
-    function initiateSettle(bytes32 sourceDomain, uint256 batchedDaiToFlush) external;
+interface TargetLike {
+    function requestMint(WormholeGUID calldata wormholeGUID, uint256 maxFee) external;
+    function settle(bytes32 sourceDomain, uint256 batchedDaiToFlush) external;
 }
 
 contract WormholeRouter {
 
     mapping (address => uint256) public wards;          // Auth
-    mapping (bytes32 => address) public bridges;        // L1 bridges for each domain
-    // TODO: the reverse mapping is not needed if the L1 bridge can pass its own domain id to router.settle()
-    mapping (address => bytes32) public domains;        // Domains for each bridge
+    mapping (bytes32 => address) public targets;        // L1 bridges for each L2 domain (or the WormholeJoin for the L1 domain)
```

Comment:
Generalize this comment as the target contract just needs to support the two methods and it doesn't matter what it is anymore.